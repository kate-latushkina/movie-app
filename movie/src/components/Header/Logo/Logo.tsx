import React from "react";
import { Grid } from "@material-ui/core";
import styled from "styled-components";
import colors from "../../../variables/colors";

const Logo: React.FC = () => {
  return (
    <LogoColor>
      <b>netflix</b>
      <span>roulette</span>
    </LogoColor>
  );
};

const LogoColor = styled(Grid)`
  color: ${colors.pink};
`;

export default Logo;
