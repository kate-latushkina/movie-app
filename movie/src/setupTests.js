// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;
global.toJson = toJson;

// Fail tests on any warning
console.error = message => {
  throw new Error(message);
};
