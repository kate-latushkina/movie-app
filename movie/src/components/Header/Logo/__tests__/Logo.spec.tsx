import React from "react";
import { shallow } from "enzyme";
import Logo from "../Logo";
import "../../../../setupTests";

describe("should render Logo component", () => {
  it("should contain bold text", () => {
    const component = shallow(<Logo />);
    const wrapper = component.find("b");
    expect(wrapper.length).toBe(1);
  });
});
