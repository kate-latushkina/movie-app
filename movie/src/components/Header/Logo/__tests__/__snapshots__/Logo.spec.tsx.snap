// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Logo component should render Logo snapshot 1`] = `
<div
  className="MuiGrid-root sc-bdfBwQ hUInJQ"
>
  <b>
    netflix
  </b>
  <span>
    roulette
  </span>
</div>
`;
